source/event.c

```diff
@@ -228,7 +228,9 @@ void eventOS_scheduler_init(void)
     }
 
     /* Init Generic timer module */
+#ifndef NS_EXCLUDE_HIGHRES_TIMER
```

Comment:
Hmm, that's a little awkward. You don't want to call this, and thus call the platform_timer code, if you're in tick mode.

Can you change ns_timer and thus take this out? Make it register the callback on the first actual call to register a timer?